### DISCUSSION TITLE
Custom build: Can't open files that contain the character: “

### DISCUSSION THREAD

**Discussion by Samhayne** [staxrip/staxrip#1142] [General]
Hello,

I opened the solution in Visual Studio Community 2022, did my own build and when I'm trying to open a video file containing a “-character I'll get the following error:

(english exception below)
![image](https://github.com/staxrip/staxrip/assets/7030482/d9dbf564-2404-48db-beeb-a852cb82261e)

```
System.ArgumentException: The output byte buffer is too small to contain the encoded data, 
encoding 'Unicode (UTF-8)' fallback 'System.Text.EncoderReplacementFallback'.
```

Doesn't happen with the downloaded build.

The exception is happening in MainForm.vb here:
`If name.ToUpperInvariant Like "VTS_0#_0.VOB" Then`

Log:

> ------------------------------ Exception ------------------------------
> 
> System.ArgumentException: Der Ausgabe-Bytepuffer ist zu klein für die codierten Daten. Codierung von "Unicode (UTF-8)" in Ausweichposition "System.Text.EncoderReplacementFallback".
> Parametername: bytes
>    bei Microsoft.VisualBasic.Strings.Asc(Char String)
>    bei Microsoft.VisualBasic.CompilerServices.LikeOperator.LigatureIndex(Char ch)
>    bei Microsoft.VisualBasic.CompilerServices.LikeOperator.CanCharExpand(Char ch, Byte[] LocaleSpecificLigatureTable, CompareInfo Comparer, CompareOptions Options)
>    bei Microsoft.VisualBasic.CompilerServices.LikeOperator.ExpandString(String& Input, Int32& Length, LigatureInfo[]& InputLigatureInfo, Byte[] LocaleSpecificLigatureTable, CompareInfo Comparer, CompareOptions Options, Boolean& WidthChanged, Boolean UseFullWidth)
>    bei Microsoft.VisualBasic.CompilerServices.LikeOperator.LikeString(String Source, String Pattern, CompareMethod CompareOption)
>    bei StaxRip.MainForm.OpenVideoSourceFiles(IEnumerable`1 files, Boolean isEncoding) in C:\Users\Samhayne\Downloads\staxrip-master\staxrip-master\Source\Forms\MainForm.vb:Zeile 1979.

What can i do?
Obviously some UTF-8 issue - but what do I have to do to make the build support it?

@Dendraspis You for sure would know?
Google only finds this for me: https://www.codeproject.com/Questions/5325859/Vb-net-default-encoding

System: Windows 11 - German.

**Comment by Dendraspis** [comment]
Hello and sorry for the late response.

I don't know if I can help, because we both use obviously the same code, so maybe it is VS/Compiler or PC/OS related. The code itself works, and I was never faced with such an issue.  🫤
